client/oci_env/utils.py

```diff
@@ -1,19 +1,50 @@
-from genericpath import isfile
 import os
-import subprocess
 import pathlib
+import subprocess
 import time
 
+from genericpath import isfile
 from urllib import request
 
+#from logzero import logger
+from oci_env.logger import logger
+
 
 def get_oci_env_path():
-    return os.environ.get("OCI_ENV_PATH", os.getcwd())
+    """This returns the root directory of the oci-env checkout."""
+
+    print(type(logger))
+
+    if OCI_ENV_PATH := os.environ.get("OCI_ENV_PATH"):
+        logger.info('USING OCI_ENV_PATH FROM ENV: {OCI_ENV_PATH}')
+        return OCI_ENV_PATH
+
+    # this is the $CHECKOUT/client/oci_env/utils.py path ...
+    path = os.path.dirname(__file__)
+
+    # use git to find the root dir ...
+    pid = subprocess.run(
+        "git rev-parse --show-toplevel",
+        cwd=path,
+        shell=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
+    if pid.returncode != 0:
+        cwd = os.getcwd()
+        logger.warning(f'USING CWD {cwd} FOR OCI_ENV_PATH BECAUSE OF GIT CMD FAILURE {pid.stdout}')
+        return cwd
+
+    gitroot = pid.stdout.decode('utf-8').strip()
+    logger.info(f'USING {gitroot} FOR OCI_ENV_PATH BASED ON GIT CMD OUTPUT')
+    return gitroot.rstrip('/') + '/'
```

Comment:
Is the path suppose to have a trailing slash?

Reply:
I would say no trailing slash. My path is set to `/Users/dnewswan/code/hub/oci_env` and there are a lot of places where the path is referenced like this: https://github.com/pulp/oci_env/blob/main/base/compose.yaml#L17

I'm not totally sure if that would work with a trailing /